### PR TITLE
Piedram/appeals 40988

### PR DESCRIPTION
**Pull request by piedram**
Resolves [APPEALS-40988](https://jira.devops.va.gov/browse/APPEALS-40988)

# Description
As a Caseflow Developer, I need to update seed data to populate the NOD column when seeding correspondences, so that we can test NOD and NON NOD correspondences

## Acceptance Criteria
- [ ] Code compiles correctly

In the create_correspondences method in db/seeds/correspondence.rb, add a new block that generates correspondences where: NOD is true and status is 'unassigned'

## Testing Plan

After running the new migration and executing the seed file for correspondence check if
- [] new column `nod` in `correspondence table` the value by default is `false`.
- [] new correspondence value is created in the  `correspondence table `  with a nod column like `true` when package_document_type_id = 15.

image (37)

image (38)

-[] status is 'unassigned'.

image (39)

### CODE REVIEW

File: app/services/correspondence_auto_assign_logger.rb [hunk 1]
```diff
@@ -44,13 +44,13 @@ def assigned(task:, started_at:, assigned_to:)
...
-      nod: correspondence.nod?,
+      nod: correspondence.nod,
       status: Constants.CORRESPONDENCE_AUTO_ASSIGNMENT.statuses.completed,
       started_at: started_at
     )
 
-    if correspondence.nod?
-      batch.num_nod_packages_assigned += 1
+    if correspondence.nod
+      @batch.num_nod_packages_assigned += 1
```
Comment (cacevesva) [inline app/services/correspondence_auto_assign_logger.rb:53]: Refactor Instance variables:
Reek Violation
  [4]:InstanceVariableAssumption: CorrespondenceAutoAssignLogger assumes too much for instance variable '@batch'

File: app/services/correspondence_auto_assign_logger.rb [hunk 2]
```diff
@@ -69,13 +69,13 @@ def no_eligible_assignees(task:, started_at:)
...
+      nod: correspondence.nod,
       status: Constants.CORRESPONDENCE_AUTO_ASSIGNMENT.statuses.error,
       started_at: started_at
     )
-
-    if correspondence.nod?
-      batch.num_nod_packages_unassigned += 1
+    
+    if correspondence.nod
+      @batch.num_nod_packages_unassigned += 1
```
Comment (cacevesva) [inline app/services/correspondence_auto_assign_logger.rb:78]: Refactor Instance variables:
Reek Violation
  [4]:InstanceVariableAssumption: CorrespondenceAutoAssignLogger assumes too much for instance variable '@batch'

**Review by cacevesva — CHANGES_REQUESTED** [review]

File: app/services/correspondence_auto_assign_logger.rb (hunk 2 above)
Comment (piedram) [reply app/services/correspondence_auto_assign_logger.rb]: Done

File: app/services/correspondence_auto_assign_logger.rb (hunk 1 above)
Comment (piedram) [reply app/services/correspondence_auto_assign_logger.rb]: Done!

File: db/schema.rb [hunk 3]
```diff
@@ -707,6 +708,7 @@
     t.index ["assigned_by_id"], name: "index_correspondences_on_assigned_by_id"
     t.index ["cmp_queue_id"], name: "index_correspondences_on_cmp_queue_id"
     t.index ["correspondence_type_id"], name: "index_correspondences_on_correspondence_type_id"
+    t.index ["package_document_type_id"], name: "index_correspondences_on_package_document_type_id"
```
Comment (cacevesva) [inline db/schema.rb:711]: Is this suppose to be there or was added in?
I think this needs to be removed.

**Review by cacevesva — APPROVED** [review]

**Pull request merged**